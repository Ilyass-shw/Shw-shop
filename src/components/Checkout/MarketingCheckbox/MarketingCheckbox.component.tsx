import styled from 'styled-components';

export const CheckboxContainer = styled.div``;

// ============= ============= ============= ============= =============

export const Checkbox = styled.input`
  border: 1px solid #d9d9d9;
  width: 18px;
  border-radius: 4px;
  box-sizing: border-box;
  height: 18px;
  cursor: pointer;
  margin: 0.3rem;
  transform: translateY(4px);
`;
export const MarketingLabel = styled.label``;
