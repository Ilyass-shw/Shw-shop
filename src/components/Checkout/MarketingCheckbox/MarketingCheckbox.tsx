import React from 'react';
import { useFormContext } from 'react-hook-form';
import {
  CheckboxContainer,
  Checkbox,
  MarketingLabel,
} from './MarketingCheckbox.component';

const MarketingCheckbox: React.FC = () => {
  const { register } = useFormContext();
  return (
    <CheckboxContainer>
      <Checkbox
        {...register('buyerAcceptMarketing')}
        type="checkbox"
        id="buyer-accept-marketing"
      />

      <MarketingLabel htmlFor="buyer-accept-marketing">
        Keep me up to date on news and exclusive offers.
      </MarketingLabel>
    </CheckboxContainer>
  );
};

export default MarketingCheckbox;
