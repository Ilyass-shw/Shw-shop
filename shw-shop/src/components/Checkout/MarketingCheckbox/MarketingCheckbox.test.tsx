import React from 'react';
import userEvent from '@testing-library/user-event';
import {
  renderWithFormProvider,
  renderWithReactHookForm,
  screen,
  waitFor,
} from '../../../test-utils/testUtils';
import MarketingCheckbox from './MarketingCheckbox';

describe('MarketingCheckbox', () => {
  it('should render', () => {
    renderWithFormProvider(<MarketingCheckbox />);
  });

  it('should match snapshot', () => {
    const { asFragment } = renderWithFormProvider(<MarketingCheckbox />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should submit correct form data', async () => {
    const saveData = jest.fn();
    renderWithReactHookForm(<MarketingCheckbox />, saveData);
    const submitButton = screen.getByTestId('testSubmitButton');
    userEvent.click(submitButton);
    await waitFor(() => {
      expect(saveData).toBeCalledTimes(1);
      expect(saveData).toBeCalledWith({ buyerAcceptMarketing: false });
    });
    userEvent.click(screen.getByRole('checkbox'));
    userEvent.click(submitButton);
    await waitFor(() => {
      expect(saveData).toBeCalledTimes(2);
      expect(saveData).toBeCalledWith({ buyerAcceptMarketing: true });
    });
  });
});
