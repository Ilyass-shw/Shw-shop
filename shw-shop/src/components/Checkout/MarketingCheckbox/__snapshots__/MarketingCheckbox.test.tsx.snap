// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingCheckbox should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa"
  >
    <input
      class="sc-gsDJrp qlOGE"
      id="buyer-accept-marketing"
      name="buyerAcceptMarketing"
      type="checkbox"
    />
    <label
      class="sc-dkPtyc"
      for="buyer-accept-marketing"
    >
      Keep me up to date on news and exclusive offers.
    </label>
  </div>
</DocumentFragment>
`;
